libbeat/cfgfile/cfgfile.go

```diff
@@ -53,13 +55,17 @@ func Read(out interface{}, path string) error {
 	if err != nil {
 		return fmt.Errorf("Failed to read %s: %v. Exiting.", path, err)
 	}
-
 	filecontent = expandEnv(filecontent)
 
-	if err = yaml.Unmarshal(filecontent, out); err != nil {
-		return fmt.Errorf("YAML config parsing failed on %s: %v. Exiting", path, err)
+	config, err := yaml.NewConfig(filecontent, ucfg.PathSep("."))
```

Comment:
PathSep is some option to be passed to Merge and Unpack. Without this option PathSep is not applied. Due to Configs being reusable and potentially in multiple configs as Sub-configs, options are not stored within the configuration for consistency reasons.